monkey/infection_monkey/master/exploiter.py

```diff
@@ -121,16 +117,15 @@ def _exploit_hosts_on_queue(
 
     def _run_all_exploiters(
         self,
-        exploiters_to_run: Sequence[PluginConfiguration],
+        exploiter_configs: Mapping[ExploiterName, Mapping],
         target_host: TargetHost,
         current_depth: int,
         servers: Sequence[str],
         results_callback: Callback,
         stop: Event,
     ):
 
-        for exploiter in interruptible_iter(exploiters_to_run, stop):
-            exploiter_name = exploiter.name
+        for exploiter_name in interruptible_iter(exploiter_configs, stop):
```

Comment:
You could maybe do

```suggestion
        for exploiter_name, exploiter_config in interruptible_iter(exploiter_configs.items(), stop):
```

Then you wouldn't need the lookup on line 143. But this is a minor nit.